### PR TITLE
[SPARK-33581][SQL][TEST] Refactor HivePartitionFilteringSuite

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

This pr refactor HivePartitionFilteringSuite.

### Why are the changes needed?

To make it easy to maintain.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala [hunk 1]
```diff
@@ -108,119 +119,147 @@ class HivePartitionFilteringSuite(version: String)
...
+      chunkValue,
+      dateValue,
+      dateStrValue, {
         case expr @ In(v, list) if expr.inSetConvertible =>
           InSet(v, list.map(_.eval(EmptyRow)).toSet)
       })
   }
 
-  test("getPartitionsByFilter: cast(ds as long) in (20170102L, 20170103L) (using INSET expression)")
+ test("getPartitionsByFilter: cast(ds as long) in (20170102L, 20170103L) (using INSET expression")
```
Comment (maropu) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:254]: nit: wrong indent

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala [hunk 2]
```diff
@@ -108,119 +119,147 @@ class HivePartitionFilteringSuite(version: String)
...
-      0 to 4,
-      "aa" :: "ab" :: "ba" :: "bb" :: Nil)
+      hValue,
+      chunkValue,
+      dateValue,
+      dateStrValue)
   }
 
-  test("getPartitionsByFilter: cast(ds as long) in (20170102L, 20170103L) (using IN expression)") {
+ test("getPartitionsByFilter: cast(ds as long) in (20170102L, 20170103L) (using IN expression)") {
```
Comment (maropu) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:231]: nit: wrong indent

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala [hunk 3]
```diff
@@ -272,93 +315,105 @@ class HivePartitionFilteringSuite(version: String)
...
       attr("chunk").in("ab", "ba") && (attr("ds").cast(StringType) > "20170102"),
-      day :: Nil)
+      dsValue,
+      hValue,
+      Seq("ab", "ba"),
+      dateValue,
+      dateStrValue)
   }
 
-  test("getPartitionsByFilter: date type pruning by metastore") {
```
Comment (maropu) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:297]: You removed this test?

**Review by maropu — APPROVED** [review]
The cleanup looks good.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala (hunk 3 above)
Comment (wangyum) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala:297]: Split it to 4 tests: https://github.com/apache/spark/blob/13bcfe2794d84d9f2211625b06b38dc8dc204bd6/sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala#L355-L395

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by wangyum** [comment]
Merged to master.

**Pull request closed without merging**